app/Http/Controllers/LessonController.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Course;
+use App\Models\Lesson;
+use App\Models\Program;
+use App\Models\User;
+use App\Services\UpdateService;
+
+class LessonController extends Controller
+{
+    public function show($slug)
+    {
+        $lesson = Lesson::with('users', 'course', 'programs')->where('slug_lesson', $slug)->firstOrFail();
+        $otherCourses = Course::inRandomOrder()->take(config('course.other_course_on_detail'))->get();
+        $tags = $lesson->course->tags;
+        UpdateService::updateFinishPrograms($lesson);
```

Comment:
trong controller tìm cách sử dụng magic method với UpdateService
[How Laravel uses magic methods](https://dev.to/nvio/laravel-greatest-trick-revealed-magic-methods-31om#:~:text=Magic%20methods%20are%20methods%20available,method%20'behind%20the%20scenes'.)

Reply:
mà UpdateService chưa tường minh về ý nghĩa